### PR TITLE
Single product page

### PR DESCRIPTION
**Pull request by rishbitsnbytes**
Added Product-Detail-Page, Routing, Dynamic routing. Updated Homepage Dynamic Routing too for Top Picks ProductSection.

Kindly review.
Live Link to  the App :-  https://petkart.vercel.app/
Live Link to the Current PR :- https://petkart-ecommerce-clj8n4bas-rishbitsnbytes.vercel.app/products

Preview :-

Product-Details-Page-Preview

### CODE REVIEW

**Review by chanchal-panpaliya — APPROVED** [review]
code looks clean , Looks Good To Merge.

File: src/pages/NotFoundPage/NotFoundPage.jsx [hunk 1]
```diff
@@ -1,7 +1,15 @@
+import "./not-found-page.css";
+import notFoundImage from "../../assets/others/404-not-found-illustration.svg";
+
 const NotFoundPage = () => {
   return (
-    <div>
+    <div className="main-not-found flex-col flex-align-center flex-justify-center gap-5">
       <h1>404, Sorry this page doesn't exist!! Try later.</h1>
+      <img
```
Comment (TejasShekar) [inline src/pages/NotFoundPage/NotFoundPage.jsx:8]: 404 page does not have the svg you have in your assets. Take care of it.

**Review by TejasShekar — APPROVED** [review]
The UI is really good. Good choice of colors.
Just one issue that you can easily take care of.
LGTM.

File: src/pages/NotFoundPage/NotFoundPage.jsx (hunk 1 above)
Comment (rishbitsnbytes) [reply src/pages/NotFoundPage/NotFoundPage.jsx:8]: It's already added in src attribute. Actually, might now be showing in live app bcoz of unmerged PR. Checked again and it's showing up now. Thanks!

**Pull request merged**